packages/artifact/src/internal/upload/upload-artifact.ts

```diff
@@ -77,30 +83,50 @@ export async function uploadArtifact(
   }
 
   // Upload zip to blob storage
-  const uploadResult = await uploadZipToBlobStorage(createArtifactResp.signedUploadUrl, zipUploadStream)
+  const uploadResult = await uploadZipToBlobStorage(
+    createArtifactResp.signedUploadUrl,
+    zipUploadStream
+  )
   if (uploadResult.isSuccess === false) {
     return {
       success: false
     }
   }
 
   // finalize the artifact
-  const finalizeArtifactResp = await artifactClient.FinalizeArtifact({
+  const finalizeArtifactReq: FinalizeArtifactRequest = {
     workflowRunBackendId: backendIds.workflowRunBackendId,
     workflowJobRunBackendId: backendIds.workflowJobRunBackendId,
     name: name,
     size: uploadResult.uploadSize!.toString()
-  })
+  }
+
+  if (uploadResult.md5Hash) {
+    finalizeArtifactReq.hash = StringValue.create({
+      value: `md5: ${uploadResult.md5Hash!}`
+    })
+  }
+
+  core.info(`Finalizing artifact upload`)
+
+  const finalizeArtifactResp = await artifactClient.FinalizeArtifact(
+    finalizeArtifactReq
+  )
   if (!finalizeArtifactResp.ok) {
     core.warning(`Failed to finalize artifact`)
     return {
       success: false
     }
   }
 
+  const artifactId = parseInt(finalizeArtifactResp.artifactId)
+  core.info(
+    `Artifact ${name}.zip successfully finalized. Artifact ID ${artifactId}`
+  )
+
   return {
     success: true,
     size: uploadResult.uploadSize,
-    id: parseInt(finalizeArtifactResp.artifactId) // TODO - will this be a problem due to the id being a bigint?
```

Comment:
re: this comment, wonder if we should wrap it in a `BigInt(finalizeArtifactResp.artifactId)`? @bethanyj28 i think changed the proto compiler options though to use strings

Reply:
Done!